### PR TITLE
Update windows packaging

### PR DESCRIPTION
**Pull request by chrisroberts**
Update the windows packaging process. Remove custom builds of ruby and curl,
replacing them with mingw builds (also removes dependency build requirements).

### CODE REVIEW

File: .ci/auto-sign-zip [hunk 1]
```diff
@@ -133,74 +133,79 @@ for file in "${entries[@]}"; do
     debug "ignoring path which does not require signing: %s" "${file}"
 done
 
+function sign_collection() {
+    unset jobs
+    declare -A jobs
+    local ref="${1}[@]"
+    local list=( "${!ref}" )
+    local b_cmd=( "${@:2:$#}" )
```
Comment (phinze) [inline .ci/auto-sign-zip:141]: A function comment explaining the arguments would be good as my eyeballs are refusing to parse this line 🤣 - I _think_ it's capturing "rest of args" yeah?

File: .ci/auto-sign-zip [hunk 2]
```diff
@@ -133,74 +133,79 @@ for file in "${entries[@]}"; do
...
-    if ! wait "${pid}"; then
-        failure "Could not sign Mach-O library file (%s)" "${file}"
-    fi
-    unset 'jobs[0]'
-    unset 'jobs[1]'
-    # compact jobs array
-    jobs=( "${jobs[@]}" )
-done
+debug "signing macho library files"
+sign_collection "macho_libs" "${signer}"
```
Comment (phinze) [inline .ci/auto-sign-zip:205]: image

File: .ci/auto-sign-zip [hunk 3]
```diff
@@ -133,74 +133,79 @@ for file in "${entries[@]}"; do
...
+    for pid in "${!jobs[@]}"; do
+        debug "waiting for pid (%d) to complete (siging %s)" "${pid}" "${jobs[$pid]}"
+        wait "${pid}" ||
+            failure "Failed to sign file (%s)" "${jobs[$pid]}"
+    done
+}
+
 # Store job information
-jobs=()
+declare -A jobs
```
Comment (phinze) [inline .ci/auto-sign-zip:195]: Do we still need this out here?

File: package/windows/build_vagrant_msi.ps1 [hunk 4]
```diff
@@ -0,0 +1,373 @@
+<#
+.SYNOPSIS
+Packages a Vagrant installer from a substrate artifact and
+an installed artifact
```
Comment (phinze) [inline package/windows/build_vagrant_msi.ps1:4]: Are these powershell scripts inherited from the old build process or is it all new?

File: substrate/windows/substrate-builder [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+        needed+=( "${t}" )
+    fi
+done
+
+if ! command -v dirname > /dev/null; then
+    needed+=( "coreutils" )
+fi
+
+if [ "${#needed[@]}" -gt 0 ]; then
+    pacman -S --noconfirm "${needed[@]}" || exit
```
Comment (phinze) [inline substrate/windows/substrate-builder:32]: so weird to see pacman in this context! but lo and behold [that's what msys2 uses](https://www.msys2.org/docs/package-management/).

File: substrate/windows/substrate-builder [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+popd > /dev/null || exit
+
+# Do work within an isolated work directory
+work_dir="$(mktemp -d vagrant-substrate.XXXXX)" || exit
+pushd "${work_dir}" > /dev/null || exit
+work_dir="$(pwd)" || exit
+
+# Grab styrene as it's what's building our substrate
+printf "checking out styrene...\n"
+git clone https://github.com/chrisroberts/styrene ./styrene  || exit
```
Comment (phinze) [inline substrate/windows/substrate-builder:70]: is this fork still required? i can't tell if there's any difference between this and upstream

**Review by phinze — APPROVED** [review]
Lots of commentary but nothing blocking! Just mostly stuff for my understanding

File: .ci/auto-sign-zip (hunk 1 above)
Comment (chrisroberts) [reply .ci/auto-sign-zip:141]: haha, yeah. i'll toss a comment in!

File: package/windows/build_vagrant_msi.ps1 (hunk 4 above)
Comment (chrisroberts) [reply package/windows/build_vagrant_msi.ps1:4]: They are new

File: .ci/auto-sign-zip (hunk 3 above)
Comment (chrisroberts) [reply .ci/auto-sign-zip:195]: We do not!

File: .ci/auto-sign-zip (hunk 1 above)
Comment (chrisroberts) [reply .ci/auto-sign-zip]: included some extra comments and a header on the function with brief explanation :slightly_smiling_face:

**Pull request merged**